### PR TITLE
board: arm: Adds mps2 an521 pyocd runner

### PR DESCRIPTION
**Pull request by ithinuel**
This PR enables `PyOCD` as a runner for mps2_an521 [Corstone-201](https://developer.arm.com/Processors/Corstone-201) reference package.

This PR depends on:

- #60780

### CODE REVIEW

File: samples/subsys/ipc/rpmsg_service/Kconfig.sysbuild [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+source "share/sysbuild/Kconfig"
+
+config RPMSG_REMOTE_BOARD
+string
+	default "nrf5340dk_nrf5340_cpunet" if $(BOARD) = "nrf5340dk_nrf5340_cpuapp"
+	default "bl5340_dvk_cpunet" if $(BOARD) = "bl5340_dvk_cpuapp"
+	default "lpcxpresso54114_m0" if $(BOARD) = "lpcxpresso54114_m4"
+	default "mps2_an521_remote" if $(BOARD) = "mps2_an521"
+	default "v2m_musca_b1_ns" if $(BOARD) = "v2m_musca_b1"
+	default "v2m_musca_b1_ns" if $(BOARD) = "v2m_musca_b1"
```
Comment (nordicjm) [inline samples/subsys/ipc/rpmsg_service/Kconfig.sysbuild:14]: duplicate

File: samples/subsys/ipc/rpmsg_service/sample.yaml [hunk 2]
```diff
@@ -11,6 +11,7 @@ tests:
       - mps2_an521
```
Comment (nordicjm) [inline samples/subsys/ipc/rpmsg_service/sample.yaml:11]: Should the allow list be updated?

File: samples/subsys/ipc/rpmsg_service/remote/sample.yaml [hunk 3]
```diff
@@ -12,3 +10,4 @@ tests:
     integration_platforms:
       - v2m_musca_b1_ns
```
Comment (nordicjm) [inline samples/subsys/ipc/rpmsg_service/remote/sample.yaml:11]: Is this really needed if both applications are now built using sysbuild?

Comment (ithinuel) [reply samples/subsys/ipc/rpmsg_service/remote/sample.yaml:11]: Maybe, I don't know well enough this part to comment.

File: samples/subsys/ipc/rpmsg_service/Kconfig.sysbuild (hunk 1 above)
Comment (ithinuel) [reply samples/subsys/ipc/rpmsg_service/Kconfig.sysbuild:14]: Oh, indeed, thank you!

File: samples/subsys/ipc/rpmsg_service/sample.yaml (hunk 2 above)
Comment (ithinuel) [reply samples/subsys/ipc/rpmsg_service/sample.yaml:11]: What is the allow list in this context?
I still need to re-run this after the holiday break but this test still passed on mps2_an521 provided it was built with sysbuild.

**Review by nordicjm — CHANGES_REQUESTED** [review]
I don't know how CI is passing because this does not work @PerMac ? Before this change I get output like this:
```
Remote core received a message: 88
Master core received a message: 89
Remote core received a message: 90
Master core received a message: 91
Remote core received a message: 92
Master core received a message: 93
Remote core received a message: 94
Master core received a message: 95
Remote core received a message: 96
Master core received a message: 97
Remote core received a message: 98
RPMsg Service demo ended.
Master core received a message: 99
RPMsg Service demo ended.
qemu-system-arm: terminating on signal 2
ninja: build stopped: interrupted by user.
```
I've built with these changes both manually and using twister, twister fails and the log shows the same output, when I run manually I get:
```
qemu-system-arm: warning: nic lan9118.0 has no peer
*** Booting Zephyr OS build zephyr-v3.5.0-2662-g6754bbef3374 ***
Starting application thread!

RPMsg Service [master] demo started
*** Booting Zephyr OS build zephyr-v3.5.0-2662-g6754bbef3374 ***
Starting application thread!

RPMsg Service [remote] demo started
qemu-system-arm: terminating on signal 2
```

So something is very wrong here

**Comment by PerMac** [comment]
> Having a look in the logs it hasn't even ran this sample on the integration platform mps2_an521

the sample did passed in the CI. 
```
INFO    -  59/720 mps2_an521                samples/subsys/ipc/rpmsg_service/sample.ipc.rpmsg_service PASSED (qemu 2.388s)
```
at build (1)
I also tested it locally with upstream's docker to confirm

**Comment by nordicjm** [comment]
Have re-tested also and it's passing, can't explain why it was failing earlier

**Comment by carlescufi** [comment]
@nordicjm can you take another look?

**Review by nordicjm — CHANGES_REQUESTED** [review]

File: samples/subsys/ipc/rpmsg_service/sysbuild.cmake [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+ExternalZephyrProject_Add(
+    APPLICATION rpmsg_service_remote
+    SOURCE_DIR ${APP_DIR}/remote
+    BOARD ${SB_CONFIG_RPMSG_REMOTE_BOARD}
+  )
+
+add_dependencies(rpmsg_service rpmsg_service_remote)
+if(("${SB_BOARD}" STREQUAL "lpcxpresso54114_m4"))
+  sysbuild_add_dependencies(CONFIGURE core_m0_inc_target rpmsg_service_remote)
+endif()
```
Comment (57300) [inline samples/subsys/ipc/rpmsg_service/sysbuild.cmake:18]: Can this be removed?
* `SB_BOARD` is undefined, so this code does nothing.
* I don't see `core_m0_inc_target` defined anywhere, neither as a CMake target nor a sysbuild image.

**Review by tejlmand — CHANGES_REQUESTED** [review]
Seems nrf5340dk_nrf5340_cpuapp is broken when building using sysbuild and this PR.

Will need some time to look further into this.

**Comment by ithinuel** [comment]
@tejlmand @nordicjm let me know if anything else needs fixing.

File: samples/subsys/ipc/rpmsg_service/remote/CMakeLists.txt [hunk 5]
```diff
@@ -1,25 +1,15 @@
...
-	OR "${BOARD}" STREQUAL "esp32s3_devkitm_appcpu")
-	message(STATUS "${BOARD} compile as slave in this sample")
-else()
-	message(FATAL_ERROR "${BOARD} was not supported for this sample")
-endif()
-
 find_package(Zephyr REQUIRED HINTS $ENV{ZEPHYR_BASE})
 project(rpmsg_service_remote)
 
+message(STATUS "${BOARD} compile as slave in this sample")
```
Comment (nordicjm) [inline samples/subsys/ipc/rpmsg_service/remote/CMakeLists.txt:12]:
```suggestion
message(STATUS "${BOARD} compile as remote in this sample")
```

**Review by nordicjm — COMMENTED** [review]
Minor nit.

**Review by tejlmand — APPROVED** [review]
Approved, the nrf53 issues seems unrelated to this PR.

Thanks for this improvement.

**Review by nordicjm — APPROVED** [review]

**Pull request merged**